### ISSUE TITLE
nano-modeline-faces vs faces

### ISSUE THREAD

**Issue by aaronjensen** [rougier/nano-modeline#54]
Hi, now that the simpler branch is merged, I've started to try to rebuild my modeline with the new toolkit. I was curious about the reason for `nano-modeline-faces` vs having separate named faces for each possible face. As a user who is needing to customize, what I have ended up doing is adding my own faces and using them. I've never had to do that before in any other package, so it's a bit cumbersome. Also, the way that the base face gets inherited in is clever, but also a bit surprising. I personally think the old way was more manageable as a consumer of the library, but I may very well be missing something.

**Comment by alkurbatov** [comment]
Hi, +1 to this issue. While I like the new branch  pretty much,  previously it was relatively simple to change foreground for secondary elements (line and column numbers), now it requires additional tinkering.

**Comment by aaronjensen** [comment]
What do you mean by "dynamic"? How is it any different from dynamically constructing the face name, e.g., `nano-modeline-name-active-face`? Isn't that what the old version did?

**Comment by rougier** [comment]
for example, this is the code for the buffer name. I just call the `nano-modeline-face` function that will take care of applying active or inactive face (using `name`as a key). How would you write this?

```emacs-lisp
(defun nano-modeline-buffer-name (&optional name)
  "Buffer name"
  
  (propertize
   (cond (name name)
         ((buffer-narrowed-p) (format"%s [narrow]" (buffer-name)))
         (t (buffer-name)))
   'face (nano-modeline-face 'name)))
```

**Comment by aaronjensen** [comment]
`(nano-modeline-face 'name)` would resolve to `nano-modeline-name-active-face` (or inactive). You would declare all permutations of faces.

I feel like I'm missing something here though? The keys aren't dynamic, right? They can't be *anything*, there's only a fixed set (at least declared by nano-modeline, obviously an end user could decide to add their own faces).

**Comment by rougier** [comment]
Yes, keys are not dynamic and they can be anything. They are used to find a corresponding face using the nano-modeline-faces and the current state (active / inactive).

Alternative would be to call nano-modeline-face with active-face and inactive face. Or to have a convention for naming faces (e.g. some-face-active some-face-inactive)

**Comment by aaronjensen** [comment]
Yeah, I think the naming convention makes the most sense. It gives users of the package a way to customize faces they are used to, and power users a utility and convention for additional faces. 

Alternatively, you can do both—maintain the map but also declare faces for all of the default faces and states. In other words, rather than using bold, you would have an active name face declared and mapped. I’m just not sure that additional level of indirection buys us anything.

**Comment by rougier** [comment]
Ok, I'll create a branch to test. Not sure when though...
